pkg/k8s/apis/cilium.io/v2alpha1/types.go

```diff
@@ -75,3 +75,44 @@ type CiliumEndpointSliceList struct {
 	// Items is a list of CiliumEndpointSlice.
 	Items []CiliumEndpointSlice `json:"items"`
 }
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:categories={cilium},singular="ciliumworldcidrset",path="ciliumworldcidrsets",scope="Cluster",shortName={cwcidr}
+// +kubebuilder:printcolumn:JSONPath=".metadata.creationTimestamp",name="Age",type=date
+// +kubebuilder:storageversion
+
+type CiliumWorldCIDRSet struct {
+	// +k8s:openapi-gen=false
+	// +deepequal-gen=false
+	metav1.TypeMeta `json:",inline"`
+	// +k8s:openapi-gen=false
+	// +deepequal-gen=false
+	metav1.ObjectMeta `json:"metadata"`
+
+	Spec CiliumWorldCIDRSetSpec `json:"spec,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:openapi-gen=false
+// +deepequal-gen=false
+
+// CiliumWorldCIDRSetList is a list of CiliumWorldCIDRSet objects.
+type CiliumWorldCIDRSetList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	// Items is a list of CiliumEgressNATPolicy.
+	Items []CiliumWorldCIDRSet `json:"items"`
+}
+
+// +kubebuilder:validation:Pattern=`^(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\/([0-9]|[1-2][0-9]|3[0-2])$`
+type CIDR string
```

Comment:
Can be replaced with 

```
+kubebuilder:validation:Format=cidr
```

Reply:
We actually might already have a CIDR type defined in the `pkg/k8s/apis/...` somewhere that can be reused here

Reply:
Yes, for example: https://github.com/cilium/cilium/blob/710297f229480bbd4fc52f39ea68a6eeb333c9d4/pkg/k8s/apis/cilium.io/v2alpha1/types.go#L15